swaggen/src/main/java/utils/io/FileMapper.java

```diff
@@ -38,14 +42,59 @@
 	 * 
 	 * @param filename the name of the YAML file
 	 * @param object the Java object
-	 * @throws JsonGenerationException
-	 * @throws JsonMappingException
-	 * @throws IOException
+	 * @param <T> The type
+	 * 
+	 * @throws JsonGenerationException JsonGenerationException
+	 * @throws JsonMappingException JsonMappingException
+	 * @throws IOException IOException
 	 */
 	public static <T> void classToYaml(String filename, T object)
 			throws JsonGenerationException, JsonMappingException, IOException {
 		ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
 		objectMapper.writeValue(new File(filename), object);
 	}
 
+	/**
+	 * Maps a Swagger object to a YAML file.
+	 * 
+	 * @param path The file path of the swagger file
+	 * @param newSwagger the Swagger
+	 * 
+	 * @throws JsonGenerationException JsonGenerationException
+	 * @throws JsonMappingException JsonMappingException
+	 * @throws IOException IOException
+	 */
+	public static void classToYaml(String path, Swagger newSwagger)
+		throws JsonGenerationException, JsonMappingException, IOException {
+		ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+		File file = new File(path.substring(1));
+
+		checkExistingEndpoints(newSwagger, file, objectMapper);
+		objectMapper.writeValue(file, newSwagger);
+	}
+
+	/**
+	 * Checks for existing endpoints in yaml files and if matched, add the existing
+	 * request methods to the new Swagger file.
+	 * @param swagger the swagger object containing all the endpoints
+	 * @param file file object, containing where it is supposed to be placed
+	 * @param objectMapper the ObjectMapper
+	 * @throws JsonGenerationException
+	 * @throws JsonMappingException
+	 * @throws IOException
+	 */
+	private static void checkExistingEndpoints(Swagger swagger, File file, ObjectMapper objectMapper) 
+		throws JsonGenerationException, JsonMappingException, IOException {
+		if(!file.exists()) {
+			return;
+		}
+		Swagger existingSwagger = objectMapper.readValue(file, Swagger.class);
+		for (String newURL: swagger.getPaths().keySet()) {
+			for (String oldURL: existingSwagger.getPaths().keySet()) {

```

Comment:
This loop should be removed. For every `newUri` we are checking if is is a key in the `exsistingSwagger` n times (where n is the size of the `exsistingSwagger` map.